vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -133,14 +136,16 @@ def parse_endpoint(entity)
       new_result = parse_base_item(entity)
       new_result[:container_groups] = []
 
-      (entity.endpoints || []).each do |endpoint|
-        next unless endpoint.targetRef.try(:kind) == 'Pod'
-        cg = @data_index.fetch_path(
-          :container_groups, :by_namespace_and_name,
-          # namespace is overriden in more_core_extensions and hence needs
-          # a non method access
-          endpoint.targetRef["table"][:namespace], endpoint.targetRef.name)
-        new_result[:container_groups] << cg unless cg.nil?
+      (entity.subsets || []).each do |subset|
+        (subset.addresses || []).each do |address|
+          next if address.targetRef.nil? || address.targetRef.kind != 'Pod'
```

Comment:
I suppose it should be (very similar to what you suggest, except for the logic):

    next if address.targetReg.try(:kind) != 'Pod'